openvasd/openvasd.c

```diff
@@ -0,0 +1,2251 @@
+/* SPDX-FileCopyrightText: 2024 Greenbone AG
+ *
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ */
+
+/**
+ * @file
+ * @brief API for Openvas Daemon communication.
+ */
+
+#include "openvasd.h"
+
+#include "../base/array.h"
+#include "../base/networking.h"
+
+#include <cjson/cJSON.h>
+#include <curl/curl.h>
+#include <curl/easy.h>
+#include <curl/multi.h>
+#include <netinet/in.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#undef G_LOG_DOMAIN
+/**
+ * @brief GLib log domain.
+ */
+#define G_LOG_DOMAIN "libgvm ovd"
+
+#define RESP_CODE_ERR -1
+#define RESP_CODE_OK 0
+
+/**
+ * @brief Struct holding the data for connecting with Openvasd.
+ */
+struct openvasd_connector
+{
+  char *ca_cert; /**< Path to the directory holding the CA certificate. */
+  char *cert;    /**< Client certificate. */
+  char *key;     /**< Client key. */
+  char *apikey;  /**< API key for authentication. */
+  char *server;  /**< original openvasd server URL. */
+  char *host;    /**< server hostname. */
+  char *scan_id; /**< Scan ID. */
+  int port;      /**< server port. */
+};
+
+/**
+ * @brief Struct holding options for Openvasd parameters.
+ */
+struct openvasd_param
+{
+  char *id;          /**< Parameter id. */
```

Comment:
why using char* here while in openvasd_credential it is gchar?

Reply:
Also wondering about this. I would expect just to use gchar everywhere. Glib is already included in openvasd.h so whoever is using the library will get gchar.